stubs/PyYAML/yaml/representer.pyi

```diff
@@ -1,60 +1,73 @@
+import datetime
 import sys
-from typing import Any
+from types import BuiltinFunctionType, FunctionType, ModuleType
+from typing import Any, ClassVar, NoReturn, Tuple, Type, TypeVar
 
-from yaml.error import YAMLError
+if sys.version_info >= (3, 0):
+    from collections.abc import Callable, Mapping, Sequence, Set
+else:
+    from typing import Callable, Mapping, Sequence, Set
+
+from yaml.error import YAMLError as YAMLError
+from yaml.nodes import MappingNode as MappingNode, Node as Node, ScalarNode as ScalarNode, SequenceNode as SequenceNode
+
+_T = TypeVar("_T")
+_N = TypeVar("_N", bound=Node)
+_R = TypeVar("_R", bound=BaseRepresenter)
 
 class RepresenterError(YAMLError): ...
 
 class BaseRepresenter:
-    yaml_representers: Any
-    yaml_multi_representers: Any
-    default_style: Any
-    default_flow_style: Any
+    yaml_representers: ClassVar[dict[Type[Any], Callable[[BaseRepresenter, Any], Node]]]
+    yaml_multi_representers: ClassVar[dict[Type[Any], Callable[[BaseRepresenter, Any], Node]]]
+    default_style: str | Any
     sort_keys: bool
-    represented_objects: Any
-    object_keeper: Any
-    alias_key: Any
-    def __init__(self, default_style=..., default_flow_style=..., sort_keys: bool = ...) -> None: ...
-    def represent(self, data): ...
+    default_flow_style: bool
+    represented_objects: dict[int, Node]
+    object_keeper: list[Any]
+    alias_key: int | Any
+    def __init__(self, default_style: str | None = ..., default_flow_style: bool = ..., sort_keys: bool = ...) -> None: ...
+    def represent(self, data) -> None: ...
+    def represent_data(self, data) -> Node: ...
     if sys.version_info < (3, 0):
         def get_classobj_bases(self, cls): ...
-    def represent_data(self, data): ...
     @classmethod
-    def add_representer(cls, data_type, representer): ...
+    def add_representer(cls: Type[_R], data_type: Type[_T], representer: Callable[[_R, _T], _N]) -> None: ...
     @classmethod
-    def add_multi_representer(cls, data_type, representer): ...
-    def represent_scalar(self, tag, value, style=...): ...
-    def represent_sequence(self, tag, sequence, flow_style=...): ...
-    def represent_mapping(self, tag, mapping, flow_style=...): ...
-    def ignore_aliases(self, data): ...
+    def add_multi_representer(cls: Type[_R], data_type: Type[_T], representer: Callable[[_R, _T], _N]) -> None: ...
```

Comment:
What's the purpose of `_N` here? Since `_N` is used only once per method, it has no effect. I assume you should just write `Node` instead of `_N`.

Reply:
@srittau you're right, good catch!